engine/script/src/script_hash.cpp

```diff
@@ -142,6 +142,57 @@ namespace dmScript
         return 1;
     }
 
+    static int CreateLuaHashUserdata(lua_State* L, dmhash_t hash)
```

Comment:
1. Create hash
2. Put it into registry
3. Save `ref` in instances
4. Put it into weak table with `(lua_Integer)hash` as a key